src/components/login/SSOLogin.tsx

```diff
@@ -0,0 +1,414 @@
+                    config: configJSON
+                },
+                active: true,
+            }
+            createSSOList(payload).then((response) => {
+                let ssoConfig = response.result;
+                this.setState({
+                    isLoading: false,
+                    saveLoading: false,
+                    ssoConfig: this.parseResponse(response.result),
+                    lastActiveSSO: {
+                        id: ssoConfig.id,
+                        name: ssoConfig.name,
+                        active: ssoConfig.active,
+                    }
+                })
+                toast.success("Saved Successful");
+            }).catch((error) => {
+                showError(error);
+                this.setState({ saveLoading: false });
+            })
+        }
+        //Update the same SSO
+        else if (this.state.lastActiveSSO) {
+            if (this.state.sso === this.state.lastActiveSSO?.name) {
+                let payload = {
+                    id: this.state.ssoConfig.id,
+                    name: this.state.sso,
+                    url: this.state.ssoConfig.url,
+                    config: {
+                        id: this.state.ssoConfig.config.id,
+                        type: this.state.ssoConfig.config.type,
+                        name: this.state.ssoConfig.config.name,
+                        config: configJSON
+                    },
+                    active: true,
+                }
+                updateSSOList(payload).then((response) => {
+                    let ssoConfig = response.result;
+                    this.setState({
+                        isLoading: false,
+                        saveLoading: false,
+                        ssoConfig: this.parseResponse(response.result),
+                        lastActiveSSO: {
+                            id: ssoConfig.id,
+                            name: ssoConfig.name,
+                            active: ssoConfig.active,
+                        }
+                    })
+                    toast.success("Saved Successful");
+                }).catch((error) => {
+                    showError(error);
+                    this.setState({ saveLoading: false });
+                })
+            }
+            else {
+                this.toggleWarningModal();
+            }
+        }
+    }
+
+    handleConfigChange(value: string): void {
+        if (this.state.configMap !== SwitchItemValues.Configuration) return;
+        try {
+            this.setState({
+                ssoConfig: {
+                    ...this.state.ssoConfig,
+                    config: {
+                        name: this.state.ssoConfig.config.name,
+                        type: this.state.ssoConfig.config.type,
+                        id: this.state.ssoConfig.config.id,
+                        config: value,
+                    },
+                }
+            });
+        } catch (error) {
+        }
```

Comment:
error handling

Reply:
try-catch not required here as there is no conversion of yaml to json. Therefore, removed it